dlext/include/SystemView.h

```diff
@@ -4,44 +4,230 @@
 #ifndef HOOMD_SYSVIEW_H_
 #define HOOMD_SYSVIEW_H_
 
+#include "DLExt.h"
+#include "hoomd/SystemDefinition.h"
 
-#include <memory>
-#include <type_traits>
+namespace dlext
+{
 
-#include "cxx11utils.h"
+namespace cxx11 = cxx11utils;
 
-#include "hoomd/ExecutionConfiguration.h"
-#include "hoomd/GlobalArray.h"
-#include "hoomd/SystemDefinition.h"
+class SystemView;
 
+// { // Aliases
 
-namespace dlext
-{
+using AccessLocation = access_location::Enum;
+const auto kOnHost = access_location::host;
+#ifdef ENABLE_CUDA
+const auto kOnDevice = access_location::device;
+#endif
 
+using AccessMode = access_mode::Enum;
+const auto kRead = access_mode::read;
+const auto kReadWrite = access_mode::readwrite;
+const auto kOverwrite = access_mode::overwrite;
 
 using ParticleDataSPtr = std::shared_ptr<ParticleData>;
 using SystemDefinitionSPtr = std::shared_ptr<SystemDefinition>;
 using ExecutionConfigurationSPtr = std::shared_ptr<const ExecutionConfiguration>;
 
+template <template <typename> class Array, typename T, typename Object>
+using ArrayPropertyGetter = const Array<T>& (Object::*)() const;
+
+template <typename T>
+using PropertyGetter = T (*)(const SystemView&, AccessLocation, AccessMode);
+
+// } // Aliases
 
 class DEFAULT_VISIBILITY SystemView {
 public:
     SystemView(SystemDefinitionSPtr sysdef);
     ParticleDataSPtr particle_data() const;
     ExecutionConfigurationSPtr exec_config() const;
     bool is_gpu_enabled() const;
+    bool in_context_manager() const;
     unsigned int local_particle_number() const;
     unsigned int global_particle_number() const;
     int get_device_id(bool gpu_flag) const;
     void synchronize();
+    void enter();
+    void exit();
```

Comment:
exit also has a very specific signature that we should use:

https://stackoverflow.com/questions/58808055/what-are-the-python-builtin-exit-argument-types
https://github.com/python/mypy/issues/4885

Reply:
Yes, you are right.
I found it. (I did see that you change the signature during the export in the first read through).

However, it does not seem to work:
At the moment
```
with Systemview(...) as sysview:
    print(sysview)
```
prints(None)

Reply:
Oh, I was missing a return statement within `__enter__`, I have added it now.

Reply:
Awesome, yes I already pulled it and that issue seems resolved.

working on it.